### PR TITLE
PLANNER-1551: Add RosterGenerator and BenchmarkApp

### PR DESCRIPTION
**Pull request by cuijulian**
- Add StringDataGenerator and RosterGenerator to generate sample data on startup with ApplicationRunner and integration tests
- Add SystemPropertiesRetriever
- Add solver config file and business constraints in Drools format 
- Add BenchmarkApp

### CODE REVIEW

File: employee-rostering-backend/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,2 +1,8 @@
+# Setting datasource
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.url=jdbc:h2:mem:testdb
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/main/resources/application.properties]: Why do we need this? Is it for the Benchmark>

File: employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java [hunk 2]
```diff
@@ -103,23 +106,23 @@ public void getNonExistentContractTest() {
 
     @Test
     public void getNonMatchingContractTest() {
-        Integer tenantId = 1;
+        Integer tenantId = 2;
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: Why did you need to change the tenantId?

File: employee-rostering-backend/src/main/resources/application.properties (hunk 1 above)
Comment (cuijulian) [reply employee-rostering-backend/src/main/resources/application.properties]: This is not needed, Spring Boot autoconfigures the DataSource. I added it to make sure h2 is used for testing purposes, but I can remove it.

File: employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java (hunk 2 above)
Comment (cuijulian) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: This is a temporary hack to get around the fact that `RosterGenerator` generates entities whose `id`'s interfere with the entities' `id`'s in the test classes (one of the Tenants created by `RosterGenerator` has `id = 1`, which is the same `id` used in this test). Once I implement Tenant CRUD I will add `createTestTenant()` methods to each test class so that each test class has its own persisted `Tenant` entity that will not interfere with `RosterGenerator`. This . Your `BaseTest` solution also gets around this problem, but with this alternative we don't have to "turn off" `RosterGenerator` for certain test classes. I'm open to discussing this, love to know what you think.
